routers/api/v1/api.go

```diff
@@ -1237,6 +1237,7 @@ func Routes() *web.Route {
 						m.Get("/statuses", repo.GetCommitStatusesByRef)
 					}, context.ReferencesGitRepo())
 				}, reqRepoReader(unit.TypeCode))
+				m.Get("/contributors", context.ReferencesGitRepo(), reqRepoReader(unit.TypeCode), repo.GetAllContributorsStats)
```

Comment:
Please keep the permission the same as which is in `routers/web/web.go`.

Reply:
I understand what you mean but I don't know how to convert permissions in `routers/web/web.go` to be used in this file. Related lines from `web.go`:

```go
m.Group("/activity", func() {
	m.Get("", repo.Activity)
	m.Get("/{period}", repo.Activity)
	m.Group("/contributors", func() {
		m.Get("", repo.Contributors)
		m.Get("/data", repo.ContributorsData)
	})
}, context.RepoRef(), repo.MustBeNotEmpty, context.RequireRepoReaderOr(unit.TypePullRequests, unit.TypeIssues, unit.TypeReleases))
```

Some suggestion maybe?

Reply:
Use `context.RequireRepoReaderOr(unit.TypePullRequests, unit.TypeIssues, unit.TypeReleases))` but not `reqRepoReader(unit.TypeCode)`?

Reply:
OK, I gave this a second look. And I remembered why I have the api. It is actually not used by my PR but I wanted to expose an endpoint to users so that they can get that information from api. Currently it is working fine. When I go to `/api/v1/repos/{owner}/{repo}/contributors`, I get a list of contributors. This works correctly when there is no code in the repo. But if change the mentioned line above based on your suggestion, I get errors. So I think it is working fine and we should not change anything.

Reply:
Maybe you can move this change to another PR after this PR merged. It will make it review easier since it's unrelated changes.

Reply:
OK, you are right. I have removed it in 3740877